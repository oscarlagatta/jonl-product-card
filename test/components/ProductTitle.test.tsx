import { create } from 'react-test-renderer';
import React from 'react';
import { ProductTitle, ProductCard } from '../../src';

import { product1 } from '../data/products';

describe('Product Title', () => {
  test('must show the component with the custom title', () => {

    const wrapper = create(
      <ProductTitle title="Custom Product" className="custom-classu"/>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();

  });

  test('Must show the component with the product name', () => {
    const wrapper = create(
      <ProductCard product={product1}>
        {
          () => (
            <ProductTitle />
          )
        }
      </ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Must show the component with the product name', () => {
    const wrapper = create(
      <ProductCard product={product1}>
        {
          () => (
            <ProductTitle />
          )
        }
      </ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });


})