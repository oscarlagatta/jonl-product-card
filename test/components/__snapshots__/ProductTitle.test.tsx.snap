// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Title Must show the component with the product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - without image 
  </span>
</div>
`;

exports[`Product Title Must show the component with the product name 2`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - without image 
  </span>
</div>
`;

exports[`Product Title must show the component with the custom title 1`] = `
<span
  className="productDescription custom-classu"
>
  Custom Product
</span>
`;
